src/EntryControllerTest

```diff
@@ -0,0 +1,337 @@
+        
+        assertEquals(mockEntry.getSTATE().toString(), "ENTERING");//verify initial state
+        
+        when(mockIS.carIsDetected()).thenReturn(false);
+        when(mockIS.getId()).thenReturn("Entry Inside Sensor");
+        mockEntry.carEventDetected("Entry Inside Sensor", false);
+        
+        assertEquals(mockEntry.getSTATE().toString(), "TAKEN");//verify transition state
+    }
+    
+    @Test
+    public void transitionTestFromENTEREDtoIDLE() {
+        //from ENTERED to IDLE
+        when(mockOS.carIsDetected()).thenReturn(true);
+        when(mockOS.getId()).thenReturn("Entry Outside Sensor");
+        mockEntry.carEventDetected("Entry Outside Sensor", true);
+        
+        when(mockCarpark.issueAdhocTicket()).thenReturn(mockTicket);
+        mockEntry.buttonPushed();
+        mockEntry.ticketTaken();
+        when(mockIS.carIsDetected()).thenReturn(true);
+        when(mockIS.getId()).thenReturn("Entry Inside Sensor");
+        mockEntry.carEventDetected("Entry Inside Sensor", true);
+        
+        when(mockOS.carIsDetected()).thenReturn(false);
+        when(mockOS.getId()).thenReturn("Entry Outside Sensor");
+        mockEntry.carEventDetected("Entry Outside Sensor", false);
+        
+        assertEquals(mockEntry.getSTATE().toString(), "ENTERED");//verify initial state
+        
+        when(mockIS.carIsDetected()).thenReturn(false);
+        when(mockIS.getId()).thenReturn("Entry Inside Sensor");
+        mockEntry.carEventDetected("Entry Inside Sensor", false);
+        
+        when(mockGate.isRaised()).thenReturn(false);
+        assertEquals(mockEntry.getSTATE().toString(), "IDLE");//verify transition state
+    }
+    
+    @Test
+    public void transitionTestFromENTEREDtoENTERING() {    
+        //from ENTERED to ENTERING
+        when(mockOS.carIsDetected()).thenReturn(true);
+        when(mockOS.getId()).thenReturn("Entry Outside Sensor");
+        mockEntry.carEventDetected("Entry Outside Sensor", true);
+        
+        when(mockCarpark.issueAdhocTicket()).thenReturn(mockTicket);
+        mockEntry.buttonPushed();
+        mockEntry.ticketTaken();
+       
+        when(mockIS.carIsDetected()).thenReturn(true);
+        when(mockIS.getId()).thenReturn("Entry Inside Sensor");
+        mockEntry.carEventDetected("Entry Inside Sensor", true);
+        
+        when(mockOS.carIsDetected()).thenReturn(false);
+        when(mockOS.getId()).thenReturn("Entry Outside Sensor");
+        mockEntry.carEventDetected("Entry Outside Sensor", false);
+        
+        assertEquals(mockEntry.getSTATE().toString(), "ENTERED");//verify initial state
+        
+        when(mockOS.carIsDetected()).thenReturn(true);
+        when(mockOS.getId()).thenReturn("Entry Outside Sensor");
+        mockEntry.carEventDetected("Entry Outside Sensor", true);
+        
+        assertEquals(mockEntry.getSTATE().toString(), "ENTERING");//verify transition state
+        
+    }
+    
+    
```

Comment:
I would recommend splitting this file out to a few components, one for 'state transition' and one for the constructor.

Reply:
resolved